ui/src/app/workflows/components/submit-workflow-panel.tsx

```diff
@@ -17,119 +17,92 @@ interface Props {
-                                this.setState({
-                                    entrypoint: selected.value,
-                                    selectedTemplate,
-                                    parameters: (selectedTemplate && selectedTemplate.inputs.parameters) || []
-                                });
-                            }}
-                        />
-                    </div>
-                    <div key='parameters' style={{marginBottom: 25}}>
-                        <label>Parameters</label>
-                        {this.state.workflowParameters.length > 0 && (
-                            <ParametersInput parameters={this.state.workflowParameters} onChange={workflowParameters => this.setState({workflowParameters})} />
-                        )}
-                        {this.state.parameters.length > 0 && <ParametersInput parameters={this.state.parameters} onChange={parameters => this.setState({parameters})} />}
-                        {this.state.workflowParameters.length === 0 && this.state.parameters.length === 0 ? (
-                            <>
-                                <br />
-                                <label>No parameters</label>
-                            </>
-                        ) : (
-                            <></>
-                        )}
-                    </div>
-                    <div key='labels' style={{marginBottom: 25}}>
-                        <label>Labels</label>
-                        <TagsInput tags={this.state.labels} onChange={labels => this.setState({labels})} />
-                    </div>
-                    <div key='submit'>
-                        <button onClick={() => this.submit()} className='argo-button argo-button--base' disabled={this.state.isSubmitting}>
-                            <i className='fa fa-plus' /> {this.state.isSubmitting ? 'Loading...' : 'Submit'}
-                        </button>
-                    </div>
-                </div>
-            </>
-        );
-    }
+    const [entrypoint, setEntrypoint] = useState<string>(workflowEntrypoint);
+    const [parameters, setParameters] = useState<Parameter[]>([]);
+    const [workflowParameters, setWorkflowParameters] = useState<Parameter[]>(JSON.parse(JSON.stringify(props.workflowParameters)));
+    const [templates] = useState<Template[]>([defaultTemplate].concat(props.templates));
+    const [labels, setLabels] = useState<string[]>(['submit-from-ui=true']);
+    const [error, setError] = useState<Error>();
+    const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
 
-    private getSelectedTemplate(entrypoint: string): Template {
-        for (const t of this.state.templates) {
-            if (t.name === entrypoint) {
-                return t;
-            }
-        }
-        return null;
-    }
+    const getSelectedTemplate = (name: string): Template | null => {
+        return templates.find(t => t.name === name) || null;
+    };
```

Comment:
```suggestion
    function getSelectedTemplate(name: string): Template | null {
        return templates.find(t => t.name === name) || null;
    }
```

same as above re: named function over anonymous